### PR TITLE
Fix audible emotes playing sounds while mob is gagged, mute, tongueless, or under a vow of silence

### LINKED ISSUE #68951: People can make sounds (laughing) while mouth closed by tape.
## Round ID:

Local Server with last updates

## Reproduction:

Close mouth with tape and try use emote *laugh. You hear laugh sound, but you shouldn't hear because of tape.

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request

Fixes #68951

Mobs can no longer play audible sound effects involving their mouth while: 

- They are are muzzled
- They are mimes who have a vow of silence
- They are tongueless
- They are mute

## Why It's Good For The Game

Less bugs and more depth to existing mechanics.

## Changelog

:cl:
fix: Fix audible emotes playing sounds while mob is gagged, mute, tongueless, or under a vow of silence.
/:cl:

### CODE REVIEW

**Comment by Wallemations** [comment]
I'd imagine you'd still be able to laugh if you don't have a tongue

**Comment by Mothblocks** [comment]
DNM until #68950 is merged, in which case the contents of this value will change subtly and `x == y` won't be enough anymore

**Comment by Mothblocks** [comment]
Yes

**Comment by MrDoomBringer** [comment]
> put on blindfold
> now unable to clap my hands

How does this PR address this

**Comment by Mothblocks** [comment]
wat

**Comment by MrDoomBringer** [comment]
Sorry sun was in my eyes
> remove tongue
> now unable to clap my hands

How does this PR address this

File: code/datums/emotes.dm [hunk 1]
```diff
@@ -265,6 +265,18 @@
  * Returns a bool about whether or not the user should play a sound when performing the emote.
  */
 /datum/emote/proc/should_play_sound(mob/user, intentional = FALSE)
+	if(emote_type == EMOTE_AUDIBLE)
+		if(!muzzle_ignore && user.is_muzzled())
+			return FALSE
+		if(HAS_TRAIT(user, TRAIT_MUTE))
+			return FALSE
```
Comment (MrDoomBringer) [inline code/datums/emotes.dm:272]: What I'm getting at is people with TRAIT_MUTE won't be able clap, as the [clap emote](https://github.com/tgstation/tgstation/blob/master/code/modules/mob/living/carbon/emote.dm#L18) is EMOTE_AUDIBLE, right? 

I might be misunderstanding something though

Comment (timothymtorres) [reply code/datums/emotes.dm:272]: Fixed with c0487d2aa46207aab8c4030d0a636354c139e4bc

**Comment by MrDoomBringer** [comment]
👍

**Comment by Mothblocks** [comment]
Now make this compliant with Mooshimi's PR which makes this a bitfield

**Pull request merged**